zanata-war/src/main/java/org/zanata/rest/service/ProjectIterationLocalesService.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2014, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.rest.service;
+
+import org.jboss.seam.annotations.In;
+import org.jboss.seam.annotations.Name;
+import org.zanata.common.LocaleId;
+import org.zanata.dao.ProjectIterationDAO;
+import org.zanata.model.HLocale;
+import org.zanata.model.HProjectIteration;
+import org.zanata.service.LocaleService;
+import org.zanata.service.impl.LocaleServiceImpl;
+
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import java.util.List;
+import java.util.Map;
+
+@Name("projectIterationLocalesService")
+@Path(ProjectIterationLocalesService.SERVICE_PATH)
+public class ProjectIterationLocalesService extends LocalesService implements ProjectIterationLocalesResource {
+    @PathParam("projectSlug")
+    String projectSlug;
+
+    @PathParam("iterationSlug")
```

Comment:
I agree that we should change from `"iteration"` to `"version"` everywhere, but we should really do it all at once - doing it piecemeal will just cause confusion.

In this case it is a sub-endpoint of the iteration resource, so it uses the path from that. If we change the path in the API package to use `"versionSlug"` in the path template, then we can change this here. I think we should do that when we define the alternative endpoint for that resource, which we can do by using a more complex path template (so that all the code that uses it will just work, except for the change of the slug name).

I think such changes are out of scope for this pull request.